lib/base/tlsutility.cpp

```diff
@@ -801,6 +812,30 @@ bool IsCaUptodate(X509* cert)
 	return !CertExpiresWithin(cert, LEAF_VALID_FOR);
 }
 
+/**
+ * Normalizes the specified certificate validity period in seconds to avoid
+ * issues with 32-bit time_t wrap-around in 2038.
+ *
+ * On 64-bit systems this function simply returns the specified value. On 32-bit systems it ensures that
+ * the current time plus the specified value never overflows the maximum value of time_t. Otherwise, the
+ * returned value is adjusted so that the sum is exactly the maximum value of time_t.
+ *
+ * @param seconds The desired validity period in seconds.
+ * @returns The normalized validity period in seconds.
+ */
+long NormalizeCertValidFor(long seconds)
+{
+	if constexpr (std::numeric_limits<time_t>::max() > std::numeric_limits<int32_t>::max()) {
+		return seconds;
+	}
+
+	constexpr time_t maxTimeT = static_cast<time_t>(std::numeric_limits<int32_t>::max());
+	if (auto rem = maxTimeT - time(nullptr); rem < seconds) {
+		return rem;
+	}
```

Comment:
That can still cause problems given that both this, and the following use both independently determine the current time. If we go from time 1234.99999 to 1235.00001 in between, this could still overflow.

Reply:
The whole function is gone now, but yes technically it would've overflowed!